packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx

```diff
@@ -0,0 +1,265 @@
+    const preResults: Array<IApiProperty> = [];
+
+    const { jsonDocs } = this.props;
+
+    if (jsonDocs) {
+      const propsName: string = `I${jsonDocs.name}Props`;
+      for (let j = 0; j < jsonDocs.tables.length; j++) {
+        switch (jsonDocs.tables[j].kind) {
+          case 'enum': {
+            preResults.push(this._generateEnumProperty(jsonDocs.tables[j]));
+            break;
+          }
+          case 'interface': {
+            const interfaceProperty = this._generateInterfaceProperty(jsonDocs.tables[j]);
+
+            // to ensure that I{componentName}Props comes first
+            if (propsName === jsonDocs.tables[j].name) {
+              iComponentProps.push(interfaceProperty);
+            } else {
+              preResults.push(interfaceProperty);
+            }
+
+            break;
+          }
+          case 'class': {
+            preResults.push(this._generateClassProperty(jsonDocs.tables[j]));
+            break;
+          }
+        }
+      }
+    }
+
+    return [...iComponentProps, ...preResults];
+  }
+
+  private _generateEnumProperty(table: ITableJson): IApiProperty {
+    const enumMembers: IApiEnumProperty[] = [];
+
+    const members: IEnumTableRowJson[] = table.members as IEnumTableRowJson[];
+    for (let k = 0; k < members.length; k++) {
+      // each member within the enum
+      enumMembers.push({
+        description: members[k].description,
+        name: members[k].name,
+        value: members[k].value
+      });
+    }
+
+    // the enum
+    return {
+      propertyName: table.name,
+      description: table.description,
+      title: table.kind ? table.name + ' ' + table.kind : table.name,
+      propertyType: PropertyType.enum,
+      property: enumMembers
+    };
+  }
+
+  private _generateInterfaceProperty(table: ITableJson): IApiProperty {
+    const interfaceMembers: IApiInterfaceProperty[] = [];
+
+    const members: ITableRowJson[] = table.members as ITableRowJson[];
+    for (let k = 0; k < members.length; k++) {
+      // each member within the interface
+      interfaceMembers.push({
+        description: members[k].description,
+        name: members[k].name,
+        typeTokens: members[k].typeTokens,
+        deprecated: members[k].deprecated,
+        defaultValue: members[k].defaultValue || ''
+      });
+    }
+
+    // the interface
+    return {
+      propertyName: table.name,
+      description: table.description,
+      extendsTokens: table.extendsTokens,
+      title: table.kind ? table.name + ' ' + table.kind : table.name,
+      propertyType: PropertyType.interface,
+      property: interfaceMembers
+    };
+  }
+
+  private _generateClassProperty(table: ITableJson): IApiProperty {
+    // class members are a mix of IApiInterfaceProperty and IMethod
+    const classMembers: IApiInterfaceProperty[] = [];
+    const classMethods: IMethod[] = [];
+
+    const members: ITableRowJson[] = table.members as ITableRowJson[];
```

Comment:
These casts can probably be removed with some refactoring, but we can do that separately.

Reply:
Will keep this in mind for the next PR!